src/widgets/paragraph.rs

```diff
@@ -136,21 +136,28 @@ where
         let mut line_composer: Box<dyn LineComposer> = if self.wrapping {
             Box::new(WordWrapper::new(&mut styled, text_area.width))
         } else {
-            Box::new(LineTruncator::new(&mut styled, text_area.width))
+            Box::new(LineTruncator::new(
+                &mut styled,
+                text_area.width,
+                match self.alignment {
+                    Alignment::Left => self.scroll.1, // only support Alignment::Left
+                    _ => 0,
+                },
+            ))
         };
         let mut y = 0;
         while let Some((current_line, current_line_width)) = line_composer.next_line() {
-            if y >= self.scroll {
+            if y >= self.scroll.0 {
                 let mut x = get_line_offset(current_line_width, text_area.width, self.alignment);
                 for Styled(symbol, style) in current_line {
-                    buf.get_mut(text_area.left() + x, text_area.top() + y - self.scroll)
-                        .set_symbol(symbol)
+                    buf.get_mut(text_area.left() + x, text_area.top() + y - self.scroll.0)
+                        .set_symbol(if symbol.is_empty() { " " } else { symbol })
```

Comment:
Why was that change required ?

Reply:
Thanks for your reviews. If the symbol is empty, the last char which rendered last time will leave on the line. May be there are other better solutions.

Reply:
No that seems right but could you add a comment to explain this. It will make the code easier to follow next time somebody has to read this part of the code :).

Reply:
Agree with you.